controllers/configurationpolicy_controller.go

```diff
@@ -1546,33 +1572,58 @@ func (r *ConfigurationPolicyReconciler) determineDesiredObjects(
 			if tmplResolver != nil && hasTemplate && desiredObj == nil { //nolint:gocritic
 				r.processedPolicyCache.Delete(plc.GetUID())
 
-				var templateContext interface{}
+				var templateContext any
 
 				// Only populate context variables as they are available:
-				// - Namespaced object with metadata.name or objectSelector
-				switch {
-				case name != "" && ns != "":
-					templateContext = struct {
-						ObjectNamespace string
-						ObjectName      string
-					}{ObjectNamespace: ns, ObjectName: name}
-
-				case name != "":
-					// - Cluster-scoped object with metadata.name or objectSelector
-					templateContext = struct {
-						ObjectName string
-					}{ObjectName: name}
-
-				case ns != "":
-					// - Unnamed namespaced object
-					templateContext = struct {
-						ObjectNamespace string
-					}{ObjectNamespace: ns}
+				// - Existing object was fetched
+				if obj.Object != nil {
+					// Remove managedFields because it has a key that's just a dot,
+					// which is problematic in the template library
+					unstructured.RemoveNestedField(obj.Object, "metadata", "managedFields")
```

Comment:
This seems reasonable anyway, since that field is currently by default hidden from `kubectl get`